### PR TITLE
gpu: Reorganize BDA used in shader

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Currently we do our updating at `vkCmdBindDescriptorSets` time, but there are things we care about at a per-draw time

This tries and removes the Post Processing from the "What descriptor type is it".

Trying to find a balance of sharing logic, but also keeping things separated to allow different parts of GPU-AV to be more efficient/accurate where needed

### CODE REVIEW

File: layers/gpu/resources/gpuav_shader_resources.h [hunk 1]
```diff
@@ -50,18 +50,18 @@ struct DescriptorCommandBinding {
...
-// Not all GPU-AV code uses each, but group together for ease of managing the memory
-struct DescriptorSetRecord {
-    // The type information will change with UpdateAfterBind so will need to update this before submitting the to the queue
-    VkDeviceAddress ds_type;  // input data
-
-    VkDeviceAddress descriptor_index_post_process;  // output data
-};
-
 // Shared among all Descriptor Indexing GPU-AV checks (so we only have to create a single buffer)
 struct DescriptorStateSSBO {
```
Comment (spencer-lunarg) [inline layers/gpu/resources/gpuav_shader_resources.h:54]: main idea behind this change, the idea is all we are passing in our BDA pointer addresses and no shader needed both `ds_type` and `descriptor_index_post_process` so they can just be their own arrays (and might even just be moved out as different bindings in the future)

File: layers/gpu/resources/gpuav_subclasses.h [hunk 2]
```diff
@@ -50,9 +50,18 @@ struct DebugPrintfBufferInfo {
...
 class CommandBuffer : public vvl::CommandBuffer {
   public:
-    // per vkCmdBindDescriptorSet() state
+    // One item per vkCmdBindDescriptorSet() called
+    // We really need this information at the draw/dispatch/action, but assume most apps will either
+    //  1. Update once and only change pipelines, so this information doesn't change between draws
+    //  2. Change once per draw, then this would be the same length as doing this per-draw
+    //
+    // Note: If the app calls vkCmdBindDescriptorSet 10 times to set descriptor set [0, 9] one at a time instead of setting [0, 9]
+    // in a single vkCmdBindDescriptorSet call than this will allocate a lot of redundant memory
```
Comment (jeremyg-lunarg) [inline layers/gpu/resources/gpuav_subclasses.h:59]: s/than/then/

**Review by jeremyg-lunarg — APPROVED** [review]

File: layers/gpu/resources/gpuav_subclasses.h (hunk 2 above)
Comment (spencer-lunarg) [reply layers/gpu/resources/gpuav_subclasses.h:59]: applied fix to https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8890

**Comment by spencer-lunarg** [comment]
> CI Vulkan-ValidationLayers build # 18059 failed.

Unrelated CI failure on Android

**Pull request merged**